File: Hash/349.两个数组的交集.cpp
/*
 * @lc app=leetcode.cn id=349 lang=cpp
 *
 * [349] 两个数组的交集
 */

// @lc code=start
class Solution
{
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
    {
        // 保存交集的结果
        unordered_set<int> resultSet;
        unordered_set<int> targetSet(nums1.begin(), nums1.end()); // nums1转化为unordered_set
        for (int num : nums2)
        {
            // find方法找到的话返回一个指向该元素的正向迭代器；反之，则返回一个指向容器中最后一个元素之后位置的迭代器
            if (targetSet.find(num) != targetSet.end())
            {
                resultSet.insert(num);
            }
        }

        return vector<int>(resultSet.begin(), resultSet.end());
    }
};
// @lc code=end

File: Hash/349.两个数组的交集.ts
/*
 * @lc app=leetcode.cn id=349 lang=typescript
 *
 * [349] 两个数组的交集
 */

// @lc code=start
function intersection(nums1: number[], nums2: number[]): number[] {
  let result = {}
  let arr = []
  for (let i = 0; i < nums1.length; i++) {
    result[nums1[i]] = 1
  }

  for (let i = 0; i < nums2.length; i++) {
    if (result[nums2[i]] === 1) {
      result[nums2[i]]++
    }
  }

  for (let key in result) {
    if (result[key] === 2) {
      arr.push(key)
    }
  }

  return arr
};
// @lc code=end

